python/baseline/services.py

```diff
@@ -58,62 +58,31 @@ def batch_input(self, tokens):
         :param tokens: tokens in format List[str] or List[List[str]]
         :return: List[List[str]]
         """
-        mxlen = 0
-        mxwlen = 0
-        vmxlen, vmxwlen = self.get_vectorizer_lens()
         # If the input is List[str] wrap it in list to make a batch of size one.
-        tokens_seq = (tokens,) if isinstance(tokens[0], six.string_types) else tokens
-        # Get sentence and word lengths from the batch
-        for tokens in tokens_seq:
-            if vmxlen != -1:
-                tokens = tokens[:vmxlen]
-            mxlen = max(mxlen, len(tokens))
-            for index in range(len(tokens)):
-                if vmxwlen != -1:
-                    tokens[index] = tokens[index][:vmxwlen]
-                mxwlen = max(mxwlen, len(tokens[index]))
-        return tokens_seq, mxlen, mxwlen
-
-    def set_vectorizer_lens(self, mxlen, mxwlen):
-        """Set the max lengths on the vectorizers if unset.
-
-        :param mxlen: `int`: The max length of an example
-        :param mxwlen: `int`: The max length of a word in the batch
-        """
-        for k, vectorizer in self.vectorizers.items():
-            if hasattr(vectorizer, 'mxlen'):
-                if self.preproc == 'client':
-                    vectorizer.mxlen = mxlen
-                elif self.preproc == 'server' and vectorizer.mxlen == -1:  # keep the original behavior
-                    vectorizer.mxlen = mxlen
-            if hasattr(vectorizer, 'mxwlen'):
-                if self.preproc == 'client':
-                    vectorizer.mxwlen = mxwlen
-                elif self.preproc == 'server' and vectorizer.mxwlen == -1:  # keep the original behavior
-                    vectorizer.mxwlen = mxwlen
-
-    def get_vectorizer_lens(self):
-        """get the max len from the vectorizers if defined
-        """
-        mxlen = -1
-        mxwlen = -1
-        if self.preproc == 'server' and self.vectorizers is not None:
-            for k, vectorizer in self.vectorizers.items():
-                if hasattr(vectorizer, 'mxlen'):
-                    mxlen = vectorizer.mxlen
-                if hasattr(vectorizer, 'mxwlen'):
-                    mxwlen = vectorizer.mxwlen
-        return mxlen, mxwlen
+        tokens_batch = (tokens,) if isinstance(tokens[0], six.string_types) else tokens
+        return tokens_batch
 
-    def vectorize(self, tokens_seq):
+    def prepare_vectorizers(self, tokens):
```

Comment:
I think this method is doing to much. It is batching the input and then preparing the vectorizers. This method should probably take in the `token_batch` instead of creating it. so more like

```python
def prepare_vectorizers(self, token_batch):
     for vectorizer in self.vectorizers.values():
            vectorizer.reset()
            for tokens in tokens_batch:
                _ = vectorizer.count(tokens)
```

Then use it later with

```python
token_batch = self.batch_input(tokens)
self.prepare_vectorizers(self, token_batch)
```

Reply:
agreed